src/io/DumpH5MDParallel.cpp

```diff
@@ -0,0 +1,479 @@
+    auto group2 =
+        CHECK_HDF5(H5Gcreate(fileId, "/h5md/author", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+    auto group3 =
+        CHECK_HDF5(H5Gcreate(fileId, "/h5md/creator", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+    CHECK_HDF5(H5Gclose(group1));
+    CHECK_HDF5(H5Gclose(group2));
+    CHECK_HDF5(H5Gclose(group3));
+
+    std::vector<int> data = {2, 0};
+    CHECK_HDF5(H5LTset_attribute_int(fileId, "/h5md", "version", data.data(), data.size()));
+
+    CHECK_HDF5(H5LTset_attribute_string(fileId, "/h5md/author", "name", author.c_str()));
+
+    std::string software = "espressopp";
+    CHECK_HDF5(H5LTset_attribute_string(fileId, "/h5md/creator", "name", software.c_str()));
+
+    std::string version = Version().info();
+    CHECK_HDF5(H5LTset_attribute_string(fileId, "/h5md/creator", "version", version.c_str()));
+}
+
+void DumpH5MDParallel::writeBox(hid_t fileId)
+{
+    std::string groupName = "/particles/atoms/box";
+    auto group =
+        CHECK_HDF5(H5Gcreate(fileId, groupName.c_str(), H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+    std::vector<int> dims = {3};
+    CHECK_HDF5(
+        H5LTset_attribute_int(fileId, groupName.c_str(), "dimension", dims.data(), dims.size()));
+
+    auto boundaryType = H5Tcopy(H5T_C_S1);
+    CHECK_HDF5(H5Tset_size(boundaryType, 8));
+    CHECK_HDF5(H5Tset_strpad(boundaryType, H5T_STR_NULLPAD));
+    std::vector<hsize_t> boundaryDims = {3};
+    auto space = H5Screate_simple(int_c(boundaryDims.size()), boundaryDims.data(), nullptr);
+    auto att = H5Acreate(group, "boundary", boundaryType, space, H5P_DEFAULT, H5P_DEFAULT);
+    CHECK_HDF5(H5Awrite(att, boundaryType, "periodicperiodicperiodic"));
+    CHECK_HDF5(H5Aclose(att));
+    CHECK_HDF5(H5Sclose(space));
+    CHECK_HDF5(H5Tclose(boundaryType));
+
+    std::vector<double> edges = {system_->bc->getBoxL()[0], system_->bc->getBoxL()[1],
+                                 system_->bc->getBoxL()[2]};
+    CHECK_HDF5(
+        H5LTset_attribute_double(fileId, groupName.c_str(), "edges", edges.data(), edges.size()));
+
+    CHECK_HDF5(H5Gclose(group));
+}
+
+void DumpH5MDParallel::writeId(hid_t fileId)
+{
+    using Datatype = int64_t;
+    constexpr int64_t dimensions = 1;  ///< dimensions of the property
+
+    std::string groupName = "/particles/atoms/" + idDataset;
+    auto group = H5Gcreate(fileId, groupName.c_str(), H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+
+    std::vector<Datatype> data;
+    data.reserve(numLocalParticles * dimensions);
+    for (iterator::CellListIterator cit(system_->storage->getRealCells()); !cit.isDone(); ++cit)
+    {
+        data.emplace_back(cit->id());
+    }
+    CHECK_EQUAL(int64_c(data.size()), numLocalParticles * dimensions);
+
+    std::vector<hsize_t> localDims = {1, uint64_c(numLocalParticles), dimensions};
+    std::vector<hsize_t> globalDims = {1, uint64_c(numTotalParticles), dimensions};
+
+    std::string dataset_name = "/particles/atoms/" + idDataset + "/value";
```

Comment:
You could use already defind `groupName` variable

Reply:
fixed